package NNeuralNet.NSelectFactors.NProto;

message TFactorConfig {
  optional uint32 Index = 1;
  optional double MinValue = 2 [default = 0.0];
  optional double MaxValue = 3 [default = 1.0];
  optional bool OverwriteFactorsGenInBorders = 4;
  optional double ConstValue = 5;  // For case when for some reasons we want to fill a factor by a constant
}

message TFactorsConfig {
  optional string Slice = 1;
  repeated uint32 Indexes = 2;
  repeated TFactorConfig Factors = 3;
}
